projects/jdwp/defs/command_sets/event_request.py

```diff
@@ -0,0 +1,141 @@
+
+ThreadOnlyModifier = Struct([Field("thread", Type.THREAD_ID, "Required thread")])
+
+ClassOnlyModifier = Struct([Field("clazz", Type.REFERENCE_TYPE_ID, "Required class")])
+
+ClassMatchModifier = Struct(
+    [Field("classPattern", Type.STRING, "Restricted class pattern")]
+)
+
+StepModifier = Struct(
+    [
+        Field("thread", Type.THREAD_ID, "Required thread"),
+        Field("size", IntegralType.INT, "Size of each step"),
+        Field("depth", IntegralType.INT, "Relative call stack limit"),
+    ]
+)
+
+ClassExcludeModifier = Struct(
+    [Field("classPattern", Type.STRING, "Disallowed class pattern")]
+)
+
+LocationOnlyModifier = Struct([Field("loc", Type.LOCATION, "Required location")])
+
+ExceptionOnlyModifier = Struct(
+    [
+        Field(
+            "exceptionOrNull",
+            Type.REFERENCE_TYPE_ID,
+            "Exception to report. Null (0) means report exceptions of all types.",
+        ),
+        Field("caught", Type.BOOLEAN, "True if exception was caught"),
+        Field("uncaught", Type.BOOLEAN, "True if exception was uncaught"),
+    ]
+)
+
+FieldOnlyModifier = Struct(
+    [
+        Field("declaring", Type.REFERENCE_TYPE_ID, "Type in which field is declared."),
+        Field("fieldID", IntegralType.INT, "Required field"),
+    ]
+)
+
+InstanceOnlyModifier = Struct(
+    [Field("instance", Type.OBJECT_ID, "Required 'this' object")]
+)
+
+SourceNameMatchModifier = Struct(
+    [Field("sourceNamePattern", Type.STRING, "	Required source name pattern")]
+)
+
+__SetCommand_out_modKind = Field(
+    "modKind", UnionTag(IntegralType.BYTE, *ModifierKind), "Modifier kind"
+)
+
+__SetCommand_outArray_element_modifier = TaggedUnion(
+    tag=__SetCommand_out_modKind,
+    cases={
+        ModifierKind.COUNT: CountModifier,
+        ModifierKind.CONDITIONAL: ConditionalModifier,
+        ModifierKind.THREAD_ONLY: ThreadOnlyModifier,
+        ModifierKind.CLASS_ONLY: ClassOnlyModifier,
+        ModifierKind.CLASS_MATCH: ClassMatchModifier,
+        ModifierKind.CLASS_EXCLUDE: ClassExcludeModifier,
+        ModifierKind.STEP: StepModifier,
+        ModifierKind.LOCATION_ONLY: LocationOnlyModifier,
+        ModifierKind.EXCEPTION_ONLY: ExceptionOnlyModifier,
+        ModifierKind.FIELD_ONLY: FieldOnlyModifier,
+        ModifierKind.INSTANCE_ONLY: InstanceOnlyModifier,
+        ModifierKind.SOURCE_NAME_MATCH: SourceNameMatchModifier,
+    },
+)
+
+__SetCommand_outModifierArray_length = Field(
+    "modifiers",
+    ArrayLength(IntegralType.INT),
+    "Constraints used to control the number of generated events.",
+)
+__SetCommand_out = Struct(
+    [
+        Field("eventKind", IntegralType.BYTE, "The kind of event to request."),
+        Field("suspendPolicy", IntegralType.BYTE, "The suspend policy to use."),
+        __SetCommand_outModifierArray_length,
+        Field(
+            "outModifiers",
+            Array(
+                __SetCommand_outArray_element_modifier,
```

Comment:
The type of the element of the array is a struct consisting of two fields: tag and tagged union

Reply:
I have implemented the that, however the pyre checker is failing with:
```Too many arguments [19]: Call `TaggedUnion.__init__` expects 0 positional arguments, 2 were provided.```

looking at the `TaggedUnion` dataclass:

```
@dataclass(frozen=True)
class TaggedUnion:
    """Tagged Union class type"""

    tag: Field[UnionTag[EnumT]]
    cases: Mapping[EnumT, Struct]
```

And how tagged union is used is as:
```
TaggedUnion(
    tag=__SetCommand_out_modKind,
    cases={
        ModifierKind.COUNT: CountModifier,
        ModifierKind.CONDITIONAL: ConditionalModifier,
        ModifierKind.THREAD_ONLY: ThreadOnlyModifier,
        ModifierKind.CLASS_ONLY: ClassOnlyModifier,
        ModifierKind.CLASS_MATCH: ClassMatchModifier,
        ModifierKind.CLASS_EXCLUDE: ClassExcludeModifier,
        ModifierKind.STEP: StepModifier,
        ModifierKind.LOCATION_ONLY: LocationOnlyModifier,
        ModifierKind.EXCEPTION_ONLY: ExceptionOnlyModifier,
        ModifierKind.FIELD_ONLY: FieldOnlyModifier,
        ModifierKind.INSTANCE_ONLY: InstanceOnlyModifier,
        ModifierKind.SOURCE_NAME_MATCH: SourceNameMatchModifier,
    }
)
```
Would you kindly advice on how to resolve this fail.

Reply:
Yeah, I am not sure pyre got the definition of TaggedUnion right - I believe `EnumT` is not an allowed type unless you extend `Generic[EnumT]`